### ISSUE TITLE
bug: compactor OOM in longevity-test

### ISSUE THREAD

**Issue by fuyufjh** [risingwavelabs/risingwave#11434]
### Describe the bug

https://buildkite.com/risingwave-test/longevity-kubebench/builds/548#0189bab8-3ae8-4f37-995e-76356a0d4243

<img width="1083" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/10192522/0c571f05-8679-4acc-9daf-943f1f241b2d">

### Error message/log

I checked the first time crash of compactor node, and it seems to be caused by OOM (no panic log)

https://grafana.test.risingwave-cloud.xyz/d/liz0yRCZz1/log-search-dashboard?orgId=1&var-data_source=Logging:%20test-useast1-eks-a&var-namespace=usrlngvty-20230803-092653&var-pod=benchmark-risingwave-compactor-c-7cb8dcbf8d-vshrn&var-search=&from=1691059050000&to=1691059140000

### To Reproduce

Run longevity test with

```
BENCH_SKU="medium-3cn"
CI="true"
DURATION="10h"
INTERVAL="30m"
NEXMARK_QUERIES="q1,q2,q3,q4,q5,q7,q8,q9,q10,q14,q15,q16,q17,q18,q20"
RW_VERSION="nightly-20230801"
SLACK_CHANNEL="test_notification"
STREAMING_PARALLELISM="3"
max_clusters_allowed="2"
skip_cleanup_on_failure="0"
skip_mvs_from_mvs="1"
skip_view_queries_execution="0"
skipcountmvfromsource="1"
user="eric"
```

### How did you deploy RisingWave?

longevity test pipeline

### The version of RisingWave

nightly-20230801

**Comment by lmatz** [comment]
It is a known issue https://github.com/risingwavelabs/risingwave/issues/11142
One recent fix is https://github.com/risingwavelabs/risingwave/pull/11354

**Comment by fuyufjh** [comment]
> It is a known issue #11142 One recent fix is #11354

Thanks! Rerunning now https://buildkite.com/risingwave-test/longevity-kubebench/builds/550#0189be77-98d5-4014-8e60-1d68b3ac03d9

**Comment by Li0k** [comment]
The oom problem did not recur after the introduction of the block based xor filter. Close the issue.
